lib/msf/base/sessions/command_shell.rb

```diff
@@ -421,52 +421,68 @@ def cmd_download(*args)
 
     # Check if src exists
     if !_file_transfer.file_exist?(src)
-      print_error("The target file does not exist")
+      print_error('The target file does not exist')
       return
     end
 
+    if (::File.basename(dst) != ::File.basename(src))
+      # The destination when downloading is a local file so use this system's separator
+      dst += ::File::SEPARATOR + ::File.basename(src)
+    end
+    dst_dir = ::File.dirname(dst)
+    ::FileUtils.mkdir_p(dst_dir) if dst_dir and not ::File.directory?(dst_dir)
+
     # Get file content
-    print_status("Download #{src} => #{dst}")
+    # match the output style of the Meterpreter equivalent
+    print_status("Downloading: #{src} -> #{dst}")
     content = _file_transfer.read_file(src)
 
     # Write file to local machine
     File.binwrite(dst, content)
```

Comment:
Should this not also be changed given we writing what is now local content in a variable to a local file?

```suggestion
    ::File.binwrite(dst, content)
```